TeamCode/src/main/java/org/firstinspires/ftc/teamcode/api/vision/Vision.kt

```diff
@@ -38,10 +38,36 @@ object Vision : API() {
 
         // Build the portal
         this.portal = builder.build()
+
+        // Use a different sleep function depending on whether its a linear opmode or not.
+        val sleep = if (opMode is LinearOpMode) {
+            opMode::sleep
+        } else {
+            this::opModeSleep
+        }
+
+        // Wait until the camera has started.
+        while (this.portal.cameraState != VisionPortal.CameraState.STREAMING) {
+            opMode.telemetry.addData("Camera", this.portal.cameraState)
+            opMode.telemetry.update()
+
+            sleep(50)
+        }
```

Comment:
I don't really like blocking in an `init` call, but I think its worth it to prevent us shooting ourselves in the foot in the future.